orchestration/dagster_orchestration/hca_manage/soft_delete.py

```diff
@@ -24,6 +35,92 @@ def run(arguments: Optional[list[str]] = None) -> None:
 
 
 def soft_delete(args: argparse.Namespace, host: str) -> None:
-    hca = HcaManage(environment=args.env, data_repo_client=get_api_client(host=host),
-                    project=args.project, dataset=args.dataset)
+    hca = SoftDeleteManager(environment=args.env, data_repo_client=get_api_client(host=host),
+                            project=args.project, dataset=args.dataset)
     hca.soft_delete_rows(args.path, args.target_table)
+
+
+@dataclass
+class SoftDeleteManager:
+    environment: str
+    data_repo_client: RepositoryApi
+    project: Optional[str] = None
+    dataset: Optional[str] = None
+
+    bucket_project: str = field(init=False)
+    bucket: str = field(init=False)
+
+    def __post_init__(self) -> None:
+        self.filename_template = f"sd-{self.project}-{self.dataset}-{uuid.uuid4()}-{{table}}.csv"
+        self.bucket_project = {"prod": "mystical-slate-284720",
+                               "dev": "broad-dsp-monster-hca-dev"}[self.environment]
+        self.bucket = f"broad-dsp-monster-hca-{self.environment}-staging-storage"
+
+    @cached_property
+    def gcp_creds(self) -> google.auth.credentials.Credentials:
+        return hca_google.get_credentials()
+
+    def soft_delete_rows(self, path: str, target_table: str) -> JobId:
+        with open(path, mode="rb") as rf:
+            remote_file_path = self.put_csv_in_bucket(local_file=rf, target_table=target_table)
+            job_id = self._submit_soft_delete(target_table=target_table, target_path=remote_file_path)
+            logging.info(f"Soft delete job for table {target_table} running, job id of: {job_id}")
+            return job_id
+
+    def put_csv_in_bucket(self, local_file: BinaryIO, target_table: str) -> str:
+        """
+        Puts a local file into a GCS bucket accessible by Jade.
+        :param local_file: The file to upload.
+        :param target_table: The table name with which to format the target filename.
+        :return: The gs-path of the uploaded file.
+        """
+        storage_client = storage.Client(project=self.bucket_project, credentials=self.gcp_creds)
+        bucket = storage_client.bucket(self.bucket)
+        target_filename = self._format_filename(table=target_table)
+        blob = bucket.blob(target_filename)
+        blob.upload_from_file(local_file)
+
+        filepath = f"gs://{self.bucket}/{target_filename}"
+        logging.info(f"Put a soft-delete file here: {filepath}")
+
+        return filepath
+
+    def get_dataset_id(self) -> str:
```

Comment:
this feels like a utility method without a clear link to this service object's stated purpose of soft deletions